passes/pmgen/peepopt_shiftmul.pmg

```diff
@@ -50,8 +50,9 @@ code
 	if (GetSize(const_factor_cnst) > 20)
 		reject;
 
-	if (GetSize(port(shift, \Y)) > const_factor)
-		reject;
+	if (shift->type.in($shift, $shiftx))
+		if (GetSize(port(shift, \Y)) > const_factor)
+			reject;
```

Comment:
The following script demonstrates that disabling this check for `$shr` shift cells yields incorrect results.

```
read_verilog <<EOT
module test(input [11:0] D, input [1:0] S, output [11:0] Y);
	assign Y = D >> (S*3);
endmodule
EOT

prep

design -save gold
peepopt
design -stash gate

design -import gold -as gold test
design -import gate -as gate test

miter -equiv -make_assert -make_outputs -ignore_gold_x -flatten gold gate miter
sat -show-public -enable_undef -prove-asserts miter
```

Afaict it's only okay to skip this check if `const_factor` is a power of two. I think in that case it should be okay to skip it for all shift cell types, not just `$shr`.

Reply:
Okay, I've reverted e2c2d78 now, and added this as a new test.